src/package/packageInstall.ts

```diff
@@ -0,0 +1,263 @@
+    } else {
+      throw e;
+    }
+  }
+
+  if (queryResult?.records?.length > 0) {
+    const record = queryResult.records[0];
+    const rssUrls = record.RemoteSiteSettings.settings.map((rss) => rss.url);
+    const cspUrls = record.CspTrustedSites.settings.map((csp) => csp.endpointUrl);
+
+    const sites = [...rssUrls, ...cspUrls];
+    if (sites.length) {
+      return sites;
+    }
+  }
+}
+
+async function getStatus(connection: Connection, installRequestId: string): Promise<PackageInstallRequest> {
+  const result = await connection.tooling.retrieve('PackageInstallRequest', installRequestId);
+
+  return result as unknown as PackageInstallRequest;
+}
+
+// internal
+async function pollStatus(
+  connection: Connection,
+  installRequestId: string,
+  options: PackageInstallOptions
+): Promise<PackageInstallRequest> {
+  let packageInstallRequest: PackageInstallRequest;
+
+  const { pollingFrequency, pollingTimeout } = options;
+  let frequency: Duration;
+  if (pollingFrequency != null) {
+    frequency = isNumber(pollingFrequency) ? Duration.milliseconds(pollingFrequency) : pollingFrequency;
+  } else {
+    frequency = Duration.milliseconds(consts.PACKAGE_INSTALL_POLL_FREQUENCY);
+  }
+
+  let timeout: Duration;
+  if (pollingTimeout != null) {
+    timeout = isNumber(pollingTimeout) ? Duration.minutes(pollingTimeout) : pollingTimeout;
+  } else {
+    timeout = Duration.minutes(consts.PACKAGE_INSTALL_POLL_TIMEOUT);
+  }
+
+  const pollingOptions: Partial<PollingClient.Options> = {
+    frequency,
+    timeout,
+    poll: async (): Promise<StatusResult> => {
+      packageInstallRequest = await getStatus(connection, installRequestId);
+      getLogger().debug(installMsgs.getMessage('packageInstallPolling', [packageInstallRequest?.Status]));
+      await Lifecycle.getInstance().emit('PackageInstallRequest:status', packageInstallRequest);
+      if (['SUCCESS', 'ERROR'].includes(packageInstallRequest?.Status)) {
+        return { completed: true, payload: packageInstallRequest };
+      }
+      return { completed: false };
+    },
+  };
+
+  const pollingClient = await PollingClient.create(pollingOptions);
+
+  try {
+    getLogger().debug(`Polling for PackageInstallRequest status. Package ID = ${installRequestId}`);
+    getLogger().debug(`Polling frequency (ms): ${pollingOptions.frequency.milliseconds}`);
+    getLogger().debug(`Polling timeout (min): ${pollingOptions.timeout.minutes}`);
+    await pollingClient.subscribe();
+    return packageInstallRequest;
+  } catch (e) {
+    const errMsg = e instanceof Error ? e.message : isString(e) ? e : 'polling timed out';
+    const error = new SfError(errMsg, 'PackageInstallTimeout');
+    error.setData(packageInstallRequest);
+    if (error.stack && e.stack) {
+      // add the original stack to this new error
+      error.stack += `\nDUE TO:\n${e.stack}`;
+    }
+    throw error;
+  }
+}
+
+export async function waitForPublish(
```

Comment:
Why is this name `waitForPublish`?

Reply:
To match the associated command flag and some of the help text around it.  At first I had it as `waitForAvailability` but changed it so it was clearer that it corresponded to the command flag.